addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -889,9 +889,19 @@ export class UserValueComponent extends Component {
         this.textContentRef = useRef("text-content");
         useEffect(
             () => {
-                const textContent = this.textContentRef.el?.textContent || "";
-                if (textContent && textContent !== this.state.textContent) {
-                    this.state.textContent = textContent;
+                if (this.props.hasFakeImageItem) {
+                    const htmlContent = this.textContentRef.el?.querySelector(".o_we_fake_img_item").outerHTML || "";
+                    if (htmlContent && htmlContent !== this.state.textContent) {
+                        const markupContent = markup(htmlContent);
+                        // Avoid converting the markup back to a string.
+                        markupContent.trim = () => markupContent;
+                        this.state.textContent = markupContent;
```

Comment:
Actually, why does this happen? After thinking a bit about it, we might want to make sure the markup is not trimmed where it currently is, in case we need to perform other string operations, we would have to void the methods here as well. What do you think?

Reply:
I tried to do that, but I found no way to "safely" determine if something is a markup.
IMO (but yours might differ), checking `.constructor.name === "Markup"` looks like an even worse hack.

Or we could go for a distinct state field `markupContent` ? And handle it differently on the other side ?